resources/idlharness.js

```diff
@@ -3437,16 +3437,52 @@ IdlNamespace.prototype.test_member_attribute = function (member)
     }.bind(this));
 };
 
+IdlNamespace.prototype.test_self = function ()
+{
+    const namespaceObject = self[this.name];
+
+    subsetTestByKey(this.name, test, () => {
+        const desc = Object.getOwnPropertyDescriptor(self, this.name);
+        assert_equals(desc.value, namespaceObject, `wrong value for ${this.name} namespace object`);
+        assert_true(desc.writable, "namespace object should be writable");
+        assert_false(desc.enumerable, "namespace object should not be enumerable");
+        assert_true(desc.configurable, "namespace object should be configurable");
+        assert_false("get" in desc, "namespace object should not have a getter");
+        assert_false("set" in desc, "namespace object should not have a setter");
+    }, `${this.name} namespace: property descriptor`);
+
+    subsetTestByKey(this.name, test, () => {
+        assert_true(Object.isExtensible(namespaceObject));
+    }, `${this.name} namespace: [[Extensible]] is true`);
+
+    subsetTestByKey(this.name, test, () => {
+        assert_true(namespaceObject instanceof Object);
+        assert_equals(Object.getPrototypeOf(namespaceObject), Object.prototype);
+    }, `${this.name} namespace: [[Prototype]] is Object.prototype`);
+
+    subsetTestByKey(this.name, test, () => {
+        assert_equals(typeof namespaceObject, "object");
+        assert_true(typeof namespaceObject === "object");
+        assert_false(typeof namespaceObject === "function");
```

Comment:
These three asserts are redundant. Just do the first one.

Reply:
Safari 13.1 returns `true` for `typeof document.all === "function"`, while `"undefined"` for `typeof document.all` because of peephole optimization.
These asserts are certainly redundant here, removed them in b019bd4.